File: 240419/바구니 안의 사탕 2/candy-in-the-basket-2.py
N, K = map(int, input().split())
candies_info = [0] * 101
for _ in range(N):
    candies, pos = map(int, input().split())
    candies_info[pos] += candies

answer = 0
for i in range(len(candies_info)):
    max_candies = 0
    for j in range(i - K, i + K + 1):
        if 0 <= j < len(candies_info):
            max_candies += candies_info[j]

    answer = max(answer, max_candies)

print(answer)

File: 260222/바구니 안의 사탕 2/candy-in-the-basket-2.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int N, K;
    cin >> N >> K;

    vector<int> candies_info(101, 0);
    for (int i = 0; i < N; i++) 
    {
        int candies, pos;
        cin >> candies >> pos;
        candies_info[pos] += candies;
    }

    // Please write your code here.
    int answer = 0;
    for (int i = 0; i < (int)candies_info.size(); i++) 
    {
        int sum = 0;
        for (int j = i - K; j <= i + K; j++) 
        {
            if (0 <= j && j < (int)candies_info.size()) 
            {
                sum += candies_info[j];
            }
        }
        answer = max(answer, sum);
    }

    cout << answer << "\n";
    return 0;
}